src/elf.c

```diff
@@ -290,13 +290,29 @@ bool elf_load(elf_t *e, riscv_t *rv, memory_t *mem)
     return true;
 }
 
+#ifdef FUZZER
+bool elf_open(elf_t *e, const char *path, const uint8_t *data, size_t len)
```

Comment:
Why did you change the function prototype?

Reply:
The main purpose was to input the binary string from the fuzzer directly instead of having to go through a temporary file. 

I had an implementation with `fmemopen`, which avoided the temporary file, but not sure if it's a better approach, as it will still require changes to the function prototype.

Reply:
So I did a refactoring on the approach.

- The entire logic is extracted from the main function called `rv_init_and_execute_elf`. The function prototype is added to `riscv.h`, as adding to `riscv-private.h` will cause the error function symbol not found during linking.
- `elf_open()` signature is still modified, as we aren't generating a tmp file to store the elf output, it's hard to reuse the current code (fuzzing will be a lot of IO as the input will be mutated quickly)
-  The fuzzer will pass the input into the function `rv_init_and_execute_elf` using args. The input will then be parsed by the `parse_args`, but with a catch. As `getopt` can't deal with the input data that fuzzer generates (actually, even existing ELF files will cause the `getopt` to behave wrongly) since we have a lot of control sequence / non-ascii code points in the input. Thus, a special get opt loop is added. But the idea of sharing the same function interface by using `argc` and `args` is achieved.